### PR TITLE
Add Gitlab authentication procedure using personal tokens

### PR DESCRIPTION
**Pull request by mshaposhnik**
> Read our [Contribution guide](https://github.com/eclipse/che-docs/blob/master/CONTRIBUTING.adoc) before submitting a PR.

### What does this PR do?
Add Gitlab authentication procedure using personal tokens for provate repository factory ability.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/19170

### PR Checklist

As the author of this Pull Request I made sure that:

- [x] `vale` has been run successfully against the PR branch
- [x] Link checker has been run successfully against the PR branch
- [x] Documentation describes a scenario that is already covered by QE tests, otherwise an issue has been created and acknowledged by Che QE team
- [ ] Changed article references are updated where they are used (or a redirect has been set up on the docs side):
    - [ ] Dashboard [branding.json](https://github.com/eclipse/che-dashboard/blob/master/src/components/branding/branding.json)
    - [x] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)

### CODE REVIEW

File: modules/end-user-guide/nav.adoc [hunk 1]
```diff
@@ -21,7 +21,7 @@
 ** xref:importing-kubernetes-applications-into-a-workspace.adoc[]
 ** xref:remotely-accessing-workspaces.adoc[]
 ** xref:mounting-a-secret-as-a-file-or-an-environment-variable-into-a-workspace-container.adoc[]
-** xref:authentication-against-bitbucket-server-with-the-personal-access-token.adoc[]
+** xref:authentication-against-bitbucket-or-gitlab-server-with-the-personal-access-token.adoc[]
```
Comment (skabashnyuk) [inline modules/end-user-guide/nav.adoc]: authentication-against-bitbucket-or-gitlab-server-with-the-personal-access-token.adoc->authentication-against-scm-server-with-the-personal-access-token.adoc

File: modules/end-user-guide/partials/proc_configuring_gitlab_authentication.adoc [hunk 2]
```diff
@@ -0,0 +1,74 @@
+// configuring-gitlab-authentication
+
+[id="configuring_gitlab_authentication_{context}"]
+= Authenticating on Gitlab servers
+
+Configuring authentication on Gitlab system is a very similar to Bitbucket.
```
Comment (skabashnyuk) [inline modules/end-user-guide/partials/proc_configuring_gitlab_authentication.adoc]: replce `Gitlab` -> `GitLab`

File: modules/end-user-guide/partials/proc_configuring_gitlab_authentication.adoc [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+| Data entry
+| `token`
+| Base-64 encoded value of the personal access token
+
+|===
+
+NOTE: Encoding a string into the base64 format using the `base64` tool on Linux machines leads to adding the newline character to the end of the source string and causing a value to be unusable as the authentication header value after decoding. Avoid this by using `base64 -w0`, which removes newly added lines, or strip newlines explicitly using`tr -d \\n`.
+
+. To obtain a user ID from a secret take a look into user profile page on Gitlab web UI or make a call to a REST API URL:
```
Comment (skabashnyuk) [inline modules/end-user-guide/partials/proc_configuring_gitlab_authentication.adoc]: ` secret take ` ->  `secret, take`

File: modules/end-user-guide/partials/proc_configuring_gitlab_authentication.adoc [hunk 4]
```diff
@@ -0,0 +1,74 @@
+// configuring-gitlab-authentication
+
+[id="configuring_gitlab_authentication_{context}"]
+= Authenticating on Gitlab servers
+
+Configuring authentication on Gitlab system is a very similar to Bitbucket.
+
+Gitlab authentication is based on using personal access tokens. Each Gitlab user is able to request some
+amount of personal access tokens with different names, permissions, expiration times, and so on. Those tokens
```
Comment (skabashnyuk) [inline modules/end-user-guide/partials/proc_configuring_gitlab_authentication.adoc]: can we put a link to `GitLab` documentation?

Comment (mshaposhnik) [reply modules/end-user-guide/partials/proc_configuring_gitlab_authentication.adoc]: \+

**Review by skabashnyuk — APPROVED** [review]

**Comment by themr0c** [comment]
I am looking at the links validation failure to understand if it's legit or a regression using `htmltest`.

**Comment by themr0c** [comment]
Yes the errors are legit, can you please fix the broken links?

**Comment by MichalMaler** [comment]
Working on it

**Comment by MichalMaler** [comment]
Kinda mess in the names of included
* .adoc
* IDs
* Headings
* /pages

Will continue with this tomorrow.

**Comment by MichalMaler** [comment]
@mshaposhnik Hey Maksim, thank you for the contribution. Here is my review. Can you take a look and merge?
https://github.com/mshaposhnik/che-docs/pull/3

**Review by MichalMaler — APPROVED** [review]

File: modules/administration-guide/partials/proc_configuring-bitbucket-server-oauth1.adoc [hunk 5]
```diff
@@ -121,4 +121,4 @@ Public Key:: Paste the content of the `__<public-stripped.pub>__` file.
 * link:https://confluence.atlassian.com/bitbucketserver/personal-access-tokens-939515499.html[Bitbucket Server Personal access tokens]
 * link:https://confluence.atlassian.com/jirakb/how-to-generate-public-key-to-application-link-3rd-party-applications-913214098.html[How to generate public key to application link 3rd party applications]
 * link:https://confluence.atlassian.com/adminjiraserver/using-applinks-to-link-to-other-applications-938846918.html[Using AppLinks to link to other applications]
-* xref:end-user-guide:authentication-against-bitbucket-server-with-the-personal-access-token.adoc[].
+* xref:end-user-guide:authenticating-on-scm-server-with-a-personal-access-token.adoc[].
```
Comment (themr0c) [inline modules/administration-guide/partials/proc_configuring-bitbucket-server-oauth1.adoc:124]: The file name for the page is still  `authentication-against-scm-server-with-the-personal-access-token.adoc`, so either change the reference here, or rename the page.

Comment (mshaposhnik) [reply modules/administration-guide/partials/proc_configuring-bitbucket-server-oauth1.adoc:124]: fixed that, thanks

**Comment by mshaposhnik** [comment]
@MichalMaler @themr0c pls guys, let's finalize this. Let me know if i can do something else to merge.

**Comment by mshaposhnik** [comment]
locally i have:
```
htmltest started at 11:15:39 on build/site/che-7
========================================================================
✔✔✔ passed in 2m45.08837208s
tested 360 documents
```

**Comment by MichalMaler** [comment]
Hello Max, merge this, please. It should be fine then.
https://github.com/mshaposhnik/che-docs/pull/4

**Pull request merged**